### PR TITLE
Added private constructor in utility classes to hide implicit constructor

### LINKED ISSUE #3339: Add Private Constructors to Utility classes to hide implicit ones
As per the Sonar rule[0], Utility classes, which are collections of static members, are not meant to be instantiated. Even abstract utility classes, which can be extended, should not have public constructors.

Java adds an implicit public constructor to every class which does not define at least one explicitly. Hence, at least one non-public constructor should be defined.

You can find all classes where we need to add private constructors here[1]

[0] https://sonarcloud.io/organizations/fabric8io/rules?open=java%3AS1118&rule_key=java%3AS1118
[1] https://sonarcloud.io/organizations/fabric8io/issues?resolved=false&rules=java%3AS1118

### PR DESCRIPTION
**Pull request by MUzairS15**
## Description
Fixes #3339 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]

File: extensions/camel-k/client/src/main/java/io/fabric8/camelk/client/util/ApiVersionUtil.java [hunk 1]
```diff
@@ -30,6 +30,9 @@
    * @param apiGroup apiGroupName present if any
    * @return Just the apiGroupName part without apiGroupVersion
    */
+  private ApiVersionUtil() {
+    throw new IllegalStateException("Utility class");
+  }
```
Comment (manusa) [inline extensions/camel-k/client/src/main/java/io/fabric8/camelk/client/util/ApiVersionUtil.java:35]: This has been added in between a method and its JavaDoc, please move to line 23

File: extensions/knative/client/src/main/java/io/fabric8/knative/client/util/ApiVersionUtil.java [hunk 2]
```diff
@@ -29,6 +29,9 @@
    * @param apiGroup  apiGroupName present if any
    * @return          Just the apiGroupName part without apiGroupVersion
    */
+  private ApiVersionUtil() {
+    throw new IllegalStateException("Utility class");
+  }
```
Comment (manusa) [inline extensions/knative/client/src/main/java/io/fabric8/knative/client/util/ApiVersionUtil.java:34]: https://github.com/fabric8io/kubernetes-client/pull/3351/files#r677125649

File: extensions/service-catalog/client/src/main/java/io/fabric8/servicecatalog/client/util/ApiVersionUtil.java [hunk 3]
```diff
@@ -29,6 +29,9 @@
    * @param apiGroup  apiGroupName present if any
    * @return          Just the apiGroupName part without apiGroupVersion
    */
+  private ApiVersionUtil() {
+    throw new IllegalStateException("Utility class");
+  }
```
Comment (manusa) [inline extensions/service-catalog/client/src/main/java/io/fabric8/servicecatalog/client/util/ApiVersionUtil.java:34]: https://github.com/fabric8io/kubernetes-client/pull/3351/files#r677125649

File: extensions/tekton/client/src/main/java/io/fabric8/tekton/client/util/ApiVersionUtil.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
    * @param apiGroup  apiGroupName present if any
    * @return          Just the apiGroupName part without apiGroupVersion
    */
+   private ApiVersionUtil() {
+     throw new IllegalStateException("Utility class");
+   }
```
Comment (manusa) [inline extensions/tekton/client/src/main/java/io/fabric8/tekton/client/util/ApiVersionUtil.java:34]: https://github.com/fabric8io/kubernetes-client/pull/3351/files#r677125649

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ApiVersionUtil.java [hunk 5]
```diff
@@ -26,6 +26,9 @@
    * @param apiGroup  apiGroupName present if any
    * @return          Just the apiGroupName part without apiGroupVersion
    */
+  private ApiVersionUtil() {
+    throw new IllegalStateException("Utility class");
+  }
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ApiVersionUtil.java:31]: https://github.com/fabric8io/kubernetes-client/pull/3351/files#r677125649

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/LeaderElectionExamples.java [hunk 6]
```diff
@@ -40,7 +40,9 @@
 
 @SuppressWarnings("java:S106")
 public class LeaderElectionExamples {
-
+  private LeaderElectionExamples() {
+    throw new IllegalStateException("Utility class");
+  }
```
Comment (manusa) [inline kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/LeaderElectionExamples.java:45]: This is not a utility class. Changes don't apply here.

**Review by manusa — CHANGES_REQUESTED** [review]
Check other comments

File: kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/LeaderElectionExamples.java (hunk 6 above)
Comment (MUzairS15) [reply kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/LeaderElectionExamples.java]: done

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ApiVersionUtil.java (hunk 5 above)
Comment (MUzairS15) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ApiVersionUtil.java]: done

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**